### ISSUE TITLE
Wrong num_plurals in Catalog

### ISSUE THREAD

**Issue by denis90** [python-babel/babel#495]
# Overview Description
For `tr` and `hu` locales num_plurals is different in Catalog and plural.to_gettext.
# Steps to Reproduce

```python
from babel import Locale, plural
from babel.messages.catalog import Catalog

locale = "hu"
plural_form = Locale.parse(locale).plural_form
print "Rules: ", plural_form.rules
print "Gettext: ", plural.to_gettext(plural_form)
catalog = Catalog(locale)
print "Catalog.num_plurals: ", catalog.num_plurals
```

# Actual Results
```
Rules:  {u'one': u'n in 1'}
Gettext:  nplurals=2; plural=(((n == 1)) ? 0 : 1)
Catalog.num_plurals:  1
```
# Expected Results
```
Rules:  {u'one': u'n in 1'}
Gettext:  nplurals=2; plural=(((n == 1)) ? 0 : 1)
Catalog.num_plurals:  2
```
# Reproducibility

# Additional Information:

**Comment by nijel** [comment]
This is true for several other languages (eg. `de`, `fr`, `en` and `es`), it seems that the `other` plural form is missing for them:

```
>>> Locale.parse('en').plural_form.tags
frozenset(['one'])
>>> Locale.parse('es').plural_form.tags
frozenset(['one'])
>>> Locale.parse('de').plural_form.tags
frozenset(['one'])
>>> Locale.parse('fr').plural_form.tags
frozenset(['one'])
```

**Comment by akx** [comment]
Sorry for the super late reply – but better late than never, yeah?

We seem to currently have several definitions of plurals; the `messages` module has this [hand-maintained dictionary of plural rules](https://github.com/python-babel/babel/blob/1378a3d5e7f7f111ea7828ff7429f0abbb513700/babel/messages/plurals.py#L22-L202) and `Locale`s contain plural forms sourced from the CLDR.

In fact, it looks like list in `messages` has been [slated for removal since 2013](https://github.com/python-babel/babel/commit/09f2b54d1a923a055b2a7ac7887d2304d9c6f274#diff-612d588e27688ee612d46f4e88890de7R16).

A good course of action would seem to be to make `babel.messages.plurals.get_plural()` use the data from the `Locale` object, if it's enough to make things work like they have before.

**Comment by akx** [comment]
Also, regarding the lack of `other` tags in the `Locale`s, that seems to mean (and I should know, considering I've touched the code last in bc8f19ea1de43850c1824e405007d7be5159548e) that there are no specific rules for that tag.
